src/platform/logging/impl/stdio/Logging.cpp

```diff
@@ -4,17 +4,64 @@
 
 #include <lib/support/logging/Constants.h>
 
+#include <inttypes.h>
 #include <stdio.h>
+#include <time.h>
+
+#if defined(__APPLE__)
+#include <pthread.h>
+#include <unistd.h>
+#elif defined(__gnu_linux__)
+#include <sys/syscall.h>
+#include <unistd.h>
+#endif
 
 namespace chip {
 namespace Logging {
 namespace Platform {
 
+// TODO: investigate whether pw_chrono or pw_log could be used here
 void LogV(const char * module, uint8_t category, const char * msg, va_list v)
 {
+#if defined(_POSIX_THREAD_SAFE_FUNCTIONS)
+    flockfile(stdout);
+#endif
+
+    switch (category)
+    {
+    case kLogCategory_Error:
+        printf("\033[1;31m");
+        break;
+    case kLogCategory_Progress:
+        printf("\033[0;32m");
+        break;
+    case kLogCategory_Detail:
+        printf("\033[0;34m");
+        break;
+    }
+
+#if defined(__APPLE__) || defined(__gnu_linux__)
+    timespec ts;
+    timespec_get(&ts, TIME_UTC);
+    printf("[%lld.%06ld] ", static_cast<long long>(ts.tv_sec), static_cast<long>(ts.tv_nsec / 1000));
```

Comment:
For the Apple case this is changing the log format from "ms" to "s.us", right?  That might be OK, but we'll need to make sure there's no tooling broken by that....

Reply:
Hm yeah do we really need that granularity of timestamps? It's probably just line noise in practice

Reply:
Ok, I changed it to seconds * 1000 + milliseconds, same as it was before